### ISSUE TITLE
[CP: Stable][ios][platform_view] fix platform view clipping path intersection

### ISSUE THREAD

**Issue by kishangohel** [flutter/flutter#154712]
### Issue Link

https://github.com/flutter/flutter/issues/154249

### Target

stable

### Cherry pick PR Link

https://github.com/flutter/engine/pull/54820

### Changelog Description

Fixes issues
https://github.com/flutter/flutter/issues/153904
https://github.com/flutter/flutter/issues/154249
https://github.com/flutter/flutter/issues/154664

### Impacted Users

NA

### Impact Description

NA

### Workaround

NA

### Risk

medium

### Test Coverage

yes

**Comment by jmagman** [comment]
@hellohuanlin can you help fully fill this out?  There shouldn't be any NAs in here, and the validation steps should be filled out.

**Comment by jmagman** [comment]
> ### Cherry pick PR Link
> [flutter/engine#54820](https://github.com/flutter/engine/pull/54820)

This isn't the right PR link, and this doesn't cherry-pick cleanly due to conflicts with #54335.  I'll open a new PR.

**Comment by hellohuanlin** [comment]
Thanks. I can also do it on Monday if you haven't yet.

**Comment by nateshmbhat** [comment]
any update on this ? Eagerly waiting for this to land in stable 🤞🏻

**Comment by jmagman** [comment]
The stable PR is open at https://github.com/flutter/engine/pull/55007

**Comment by itsjustkevin** [comment]
@kishangohel I am holding this cherry-pick review as it does not currently have a valid changelog description.  Please follow our [Hotfix documentation best practices](https://github.com/flutter/flutter/blob/master/docs/releases/Hotfix-Documentation-Best-Practices.md).

**Comment by yasargil** [comment]
We are blocked by this too. Is there an estimated time when and if this lands in stable?

**Comment by hellohuanlin** [comment]
> Please follow our [Hotfix documentation best practices](https://github.com/flutter/flutter/blob/master/docs/releases/Hotfix-Documentation-Best-Practices.md).

@itsjustkevin The "Impact Description" section seems close to what's recommended in the link: "When a platform view is clipped by multiple paths, it was incorrectly clipped."
